__tests__/restore.test.ts

```diff
@@ -317,8 +165,7 @@ test("restore with cache found for restore key", async () => {
 
     expect(stateMock).toHaveBeenCalledWith("CACHE_KEY", key);
```

Comment:
Should there be a validation for "CACHE_RESULT" state as well? I know this is an existing test but if there missing validations then we can add them as we are overall re-looking at all states and outputs in this PR.

Reply:
Added test validation for this as well.